components/style/restyle_hints.rs

```diff
@@ -383,7 +415,38 @@ impl DependencySet {
     }
 
     pub fn clear(&mut self) {
-        self.deps.clear();
+        self.common_deps.clear();
+        self.attr_deps.clear();
+        self.state_deps.clear();
+    }
+}
+
+fn test_dependencies<E>(deps: &[Dependency],
```

Comment:
Also, given that we're now storing restyle hints instead of combinators, maybe "test" isn't quite the right word anymore for the operations on DependencyList. Maybe we should call it compute_partial_hint or collect_hints?